x/axelarnet/types/evm_translator.go

```diff
@@ -190,25 +190,37 @@ func ConstructWasmMessageV2(gm nexus.GeneralMessage, payload []byte) ([]byte, er
 		}
 	}
 
-	msg := wasm{
+	// When JSON unmarshalling the user payload to a map[string]interface{} type,
+	// numbers will get converted to floats. When this is marshalled again, the floats aren't converted back,
+	// leading to loss of precision, and potential non-determinism.
+	// So we leave the payload blank before the marshalling the following,
+	// and then inject the original payload into the json string instead.
+	wasmMsg := wasm{
 		Wasm: contractCall{
 			Contract:      gm.GetDestinationAddress(),
 			SourceChain:   gm.GetSourceChain().String(),
 			SourceAddress: gm.GetSourceAddress(),
-			Msg:           executeMsg,
+			Msg:           nil,
 		},
 	}
 
-	return json.Marshal(msg)
+	msg, err := json.Marshal(wasmMsg)
+	if err != nil {
+		return nil, err
+	}
+
+	msg = []byte(strings.Replace(string(msg), `"msg":null`, `"msg":`+string(payload), 1))
```

Comment:
```suggestion
	msg = []byte(strings.Replace(string(msg), `"msg":null`, fmt.Sprintf("\"msg\":%s", string(payload)), 1))
```
don't do string concat.